packages/pluggableWidgets/popup-menu-web/src/utils/document.ts

```diff
@@ -0,0 +1,262 @@
+    );
+}
+
+export function unBlockAbsoluteElementTop(
+    element: HTMLElement,
+    boundingRect: DOMRect,
+    blockingElementRect: DOMRect
+): void {
+    if (isElementBlockedTop(boundingRect, blockingElementRect)) {
+        element.style.top =
+            getPixelValueAsNumber(element, "top") + blockingElementRect.bottom - boundingRect.top + "px";
+    }
+}
+
+export function unBlockAbsoluteElementBottom(
+    element: HTMLElement,
+    boundingRect: DOMRect,
+    blockingElementRect: DOMRect
+): void {
+    if (isElementBlockedBottom(boundingRect, blockingElementRect)) {
+        element.style.top = "unset"; // Unset top defined in PopupMenu.scss
+        element.style.bottom =
+            getPixelValueAsNumber(element, "bottom") + blockingElementRect.top - boundingRect.bottom + "px";
+    }
+}
+
+export function unBlockAbsoluteElementLeft(
+    element: HTMLElement,
+    boundingRect: DOMRect,
+    blockingElementRect: DOMRect
+): void {
+    if (isElementBlockedLeft(boundingRect, blockingElementRect)) {
+        element.style.left =
+            getPixelValueAsNumber(element, "left") + blockingElementRect.right - boundingRect.left + "px";
+    }
+}
+
+export function unBlockAbsoluteElementRight(
+    element: HTMLElement,
+    boundingRect: DOMRect,
+    blockingElementRect: DOMRect
+): void {
+    if (isElementBlockedRight(boundingRect, blockingElementRect)) {
+        element.style.right =
+            getPixelValueAsNumber(element, "right") + blockingElementRect.left - boundingRect.right + "px";
+    }
+}
+
+export function getPixelValueAsNumber(element: HTMLElement, prop: keyof CSSStyleDeclaration): number {
+    const value = (getComputedStyle(element) as CSSStyleDeclaration)[prop] as string;
+    const num = Number(value.split("px")[0]);
+    return value ? num : 0;
+}
+
+export function isBehindRandomElement(
+    elementSource: HTMLElement,
+    boundingRect: DOMRect,
+    offset = 3,
+    excludeElementWithClass = ""
+): HTMLElement | false {
+    let excludeElements: HTMLElement[] = [];
+    const left = Math.round(boundingRect.left + offset);
+    const right = Math.round(boundingRect.right - offset);
+    const top = Math.round(boundingRect.top + offset);
+    const bottom = Math.round(boundingRect.bottom - offset);
+    const elementTopLeft = dynamicDocument.elementFromPoint(left, top) as HTMLElement;
+    const elementTopRight = dynamicDocument.elementFromPoint(right, top) as HTMLElement;
+    const elementBottomLeft = dynamicDocument.elementFromPoint(left, bottom) as HTMLElement;
+    const elementBottomRight = dynamicDocument.elementFromPoint(right, bottom) as HTMLElement;
+    if (excludeElementWithClass) {
+        excludeElementWithClass = excludeElementWithClass.replace(/\./g, "");
+        excludeElements = [...(dynamicDocument.querySelectorAll(`.${excludeElementWithClass}`) as any)];
+    }
+
+    if (
+        elementTopLeft &&
```

Comment:
This 3 lines are being repeated in the blocks above, what about extract this in a inner function?